File: agc003/src/bin/a.rs
use proconio::{fastout, input};

#[fastout]
fn main() {
    input! {
        s: String,
    }
    let no = s.contains('N') ^ s.contains('S') || s.contains('E') ^ s.contains('W');
    println!("{}", if !no { "Yes" } else { "No" });
}

File: agc003/a.ml
open Base;;

let s = Stdlib.read_line () in
let answer =
  let f = String.contains s in
  Bool.(f 'N' = f 'S' && f 'E' = f 'W')
in
answer |> (function true -> "Yes" | false -> "No") |> Stdlib.print_endline
